certifier_service/certlib/certlib_proofs.go

```diff
@@ -1153,6 +1239,71 @@ func VerifySevAttestation(serialized []byte, k *certprotos.KeyMessage) []byte {
 		  byte dev_public_key[PUBLIC_KEY_SIZE];
 		};
  */
+
+/*
+ * Option 1: (John prefers this option)
+ *
+ *  - Figure out how to write this fn. How to verify evidence provided by CCA-shim?
+ *  - CCA-shim provides an attest function which generates an evidence report
+ *  - Need to write something to generate an evidence package on top of that evidence.
+ *  - Write a fn that calls CCA-Attest(), packages it together before sending it to
+ *    the Cert service.
+ *
+ *    Most straightforward way to verify is to have a public-key (may be 'k' here)
+ *    and verify it. In most cases, this key is called platform-cert. This is the cert
+ *    you get for the [say, SEV] machine. This cert can be used to verify evidence
+ *    generated by [this] machine.
+ *  - For shims, we probably have one public key-cert, if cert is not provided.
+ *	Here's what's in the SDK
+ *		pub struct Report {
+ *			pub buffer: Vec<u8>,
+ *			pub user_data: Vec<u8>,
+ *		}
+ *
+ * Option 2: (Always available but we give up platform verification and trust the
+ *            SDK to do it.)
+ *
+ *  - Write Go code to call cca_Verify() externally
+ *  - Dev VerifyCCAAttestation () which will call islet_verify()
+ */
+
+//	Returns measurement
+//	serialized is the serialized cca_attestation_message
+func VerifyCCAAttestation(serialized []byte, k *certprotos.KeyMessage) []byte {
+
+	var am certprotos.CcaAttestationMessage
+	err := proto.Unmarshal(serialized, &am)
+	if err != nil {
+		fmt.Printf("VerifyCCAAttestation: Can't unmarshal SevAttestationMessage\n")
+		return nil
+	}
+
+	ptr := am.ReportedAttestation
+	if ptr == nil {
+		fmt.Printf("VerifyCCAAttestation: am.ReportedAttestation is wrong\n")
+		return nil
+	}
+
+    /*
+     * This is the hard-coded measurement provided by CCA-shim.
+     *
+	measurement := []byte {
+		0x61, 0x90, 0xEB, 0x90, 0xB2, 0x93, 0x88, 0x6C,
+		0x17, 0x2E, 0xC6, 0x44, 0xDA, 0xFB, 0x7E, 0x33,
+		0xEE, 0x2C, 0xEA, 0x65, 0x41, 0xAB, 0xE1, 0x53,
+		0x00, 0xD9, 0x63, 0x80, 0xDF, 0x52, 0x5B, 0xF9,
+	}
+    */
+	// return measurement
+    // Call the C-Go CCA verify function
+    m, err := ccaverify.CcaVerify(am.WhatWasSaid, am.ReportedAttestation)
```

Comment:
@yelvmw , @bitboom @jinbpark -- Please do review this change carefully.

I don't have enough expertise on this verification methods. I just followed the code flow / logic seen elsewhere and implemented this call to the `CcaVerify()` method.

E2E Testing shows that things seem to work .. so I'm concluding that this code is right.

Reply:
I've walked through an entire process of E2E testing for CCA. And I've drawn the same conclusion as you.
And it seems that you're using the correct hard-coded measurement in multiple places, and that measurement is equal to that inside ISLET SDK.